src/server/models/server-settings/server-settings.ts

```diff
@@ -15,111 +15,81 @@
  * limitations under the License.
  */
 
-import { BackCompat, BaseImmutable, Property, PropertyType } from "immutable-class";
 import { PluginSettings } from "../plugin-settings/plugin-settings";
 
 export type Iframe = "allow" | "deny";
 export type TrustProxy = "none" | "always";
 export type StrictTransportSecurity = "none" | "always";
 
-export interface ServerSettingsValue {
-  port?: number;
+export type ServerSettings = Readonly<ServerSettingsBase>;
+
+export interface ServerSettingsBase {
+  port: number;
   serverHost?: string;
-  serverRoot?: string;
-  serverTimeout?: number;
-  readinessEndpoint?: string;
-  livenessEndpoint?: string;
-  requestLogFormat?: string;
-  pageMustLoadTimeout?: number;
-  verbose?: boolean;
-  iframe?: Iframe;
-  trustProxy?: TrustProxy;
-  strictTransportSecurity?: StrictTransportSecurity;
-  plugins?: PluginSettings[];
+  serverRoot: string;
+  serverTimeout: number;
+  readinessEndpoint: string;
+  livenessEndpoint: string;
+  requestLogFormat: string;
+  pageMustLoadTimeout: number;
+  verbose: boolean;
+  iframe: Iframe;
+  trustProxy: TrustProxy;
+  strictTransportSecurity: StrictTransportSecurity;
+  plugins: PluginSettings[];
 }
 
-export interface ServerSettingsJS extends ServerSettingsValue {
+interface LegacyServerSettings {
```

Comment:
What is it?

Reply:
We support `healthEndpoint` property for backward compatibility. I think it should be separated to Interface